Source/HTTP/Curl/CurlEasyRequest.cpp

```diff
@@ -41,14 +41,27 @@ Result<HC_UNIQUE_PTR<CurlEasyRequest>> CurlEasyRequest::Initialize(HCCallHandle
 
     // we set both POSTFIELDSIZE and INFILESIZE because curl uses one or the
     // other depending on method
-    // We are allowing Setops to happen with a bodySize of zero to handle certain clients
+    // We are allowing Setops to happen with a bodySize of zero in linux to handle certain clients
     // not being able to handle handshakes without a fixed body size.
-    RETURN_IF_FAILED(easyRequest->SetOpt<long>(CURLOPT_POSTFIELDSIZE, static_cast<long>(bodySize)));
-    RETURN_IF_FAILED(easyRequest->SetOpt<long>(CURLOPT_INFILESIZE, static_cast<long>(bodySize)));
-
-    // read callback
-    RETURN_IF_FAILED(easyRequest->SetOpt<curl_read_callback>(CURLOPT_READFUNCTION, &ReadCallback));
-    RETURN_IF_FAILED(easyRequest->SetOpt<void*>(CURLOPT_READDATA, easyRequest.get()));
+    // The reason for an if def statement is to handle the behavioral differences in libCurl vs xCurl.
+    
+    #if HC_PLATFORM == HC_PLATFORM_LINUX
+        RETURN_IF_FAILED(easyRequest->SetOpt<long>(CURLOPT_POSTFIELDSIZE, static_cast<long>(bodySize)));
+        RETURN_IF_FAILED(easyRequest->SetOpt<long>(CURLOPT_INFILESIZE, static_cast<long>(bodySize)));
+
+        // read callback
+        RETURN_IF_FAILED(easyRequest->SetOpt<curl_read_callback>(CURLOPT_READFUNCTION, &ReadCallback));
+        RETURN_IF_FAILED(easyRequest->SetOpt<void*>(CURLOPT_READDATA, easyRequest.get()));
+    #else
+    if (bodySize > 0) {
+        RETURN_IF_FAILED(easyRequest->SetOpt<long>(CURLOPT_POSTFIELDSIZE, static_cast<long>(bodySize)));
+        RETURN_IF_FAILED(easyRequest->SetOpt<long>(CURLOPT_INFILESIZE, static_cast<long>(bodySize)));
+
+        // read callback
+        RETURN_IF_FAILED(easyRequest->SetOpt<curl_read_callback>(CURLOPT_READFUNCTION, &ReadCallback));
+        RETURN_IF_FAILED(easyRequest->SetOpt<void*>(CURLOPT_READDATA, easyRequest.get()));
+    }
+    #endif
 
```

Comment:
could you instead do:

#if HC_PLATFORM == HC_PLATFORM_GDK
// XCurl code
#else
// curl code
#endif

since HC_PLATFORM_GDK is where XCurl is used.  This is a work around for XCurl, not specific to linux